packages/shared/components/popups/MintNativeTokenConfirmationPopup.svelte

```diff
@@ -0,0 +1,120 @@
+<script lang="typescript">
+    import { localize } from '@core/i18n'
+    import { checkActiveProfileAuth } from '@core/profile'
+    import { mintNativeToken, mintTokenDetails, TokenStandard } from '@core/wallet'
+    import { closePopup, openPopup } from '@lib/popup'
+    import { Button, KeyValueBox, Text, FontWeight } from 'shared/components'
+    import { onMount } from 'svelte'
+    import { selectedAccount } from '@core/account'
+    import { handleError } from '@core/error/handlers/handleError'
+
+    export let _onMount: (..._: any[]) => Promise<void> = async () => {}
+
+    const {
+        name: tokenName,
+        totalSupply,
+        circulatingSupply,
+        decimals,
+        symbol,
+        description,
+        url,
+        logoUrl,
+        aliasId,
+    } = $mintTokenDetails
+
+    $: isTransferring = $selectedAccount.isTransferring
+
+    let detailsList: { [key in string]: { data: string; tooltipText?: string; isCopyable?: boolean } }
+    $: detailsList = {
+        ...(aliasId && {
+            alias: { data: aliasId, isCopyable: true },
+        }),
+        ...(tokenName && {
+            tokenName: { data: tokenName },
+        }),
+        ...(totalSupply && {
+            totalSupply: { data: String(totalSupply) },
+        }),
+        ...(decimals && {
+            decimals: { data: String(decimals) },
+        }),
+        ...(symbol && {
+            symbol: { data: symbol },
+        }),
+        ...(url && {
+            url: { data: url },
+        }),
+        ...(logoUrl && {
+            logoUrl: { data: logoUrl },
+        }),
+    }
+
+    async function mintAction(): Promise<void> {
+        try {
+            await mintNativeToken(Number(totalSupply), Number(circulatingSupply), {
+                standard: TokenStandard.IRC30,
+                name: tokenName,
+                symbol,
+                decimals: Number(decimals),
+                ...(description && { description }),
+                ...(url && { url }),
+                ...(logoUrl && { logoUrl }),
+            })
+            closePopup()
+        } catch (reason) {
+            handleError(reason.error)
+        }
+    }
+
+    function handleBack() {
+        closePopup()
+        openPopup({
+            type: 'mintNativeTokenForm',
+            overflow: true,
+        })
+    }
+
+    async function handleMint(): Promise<void> {
+        try {
+            await checkActiveProfileAuth(mintAction, { stronghold: true, ledger: false })
+        } catch (reason) {
+            handleError(reason.error)
+        }
+    }
+
+    onMount(async () => {
```

Comment:
I don't think onMount is allowed to be async

Reply:
it is, but its not being awaited